src/ParallelEnv.f90

```diff
@@ -1070,7 +1070,7 @@ SUBROUTINE gather_str2D_MPI_ENV_type(myPE,sendbuf,root)
     iEntry = 0
     DO j=1,UBOUND(sendbuf,DIM=2)
       DO i=1,UBOUND(sendbuf,DIM=1)
-        IF(charProc(i,j) /= rank) THEN
+        IF(charProc(i,j) /= rank .AND. charProc(i,j) >= 0) THEN
```

Comment:
If we would rather maintain the master process's data, then instead of using `-HUGE` we could keep `HUGE` and change `charProc(i,j) >= 0` to `charProc(i,j) <= myPE%nProc`. I have no real preferences either way, just thought I would mention it.

Reply:
The same two comments would apply to the the 1D version as well.

Reply:
talked offline.  We'll stick with how I did it since the checks are a bit simpler and it doesn't really make a significant difference anyways.